.github/workflows/build-test-and-sonar.yml

```diff
@@ -87,6 +87,7 @@ jobs:
           sed -i 's@/home/runner/work/power-grid-model-io/power-grid-model-io@/github/workspace@g' coverage.xml
 
       - name: SonarCloud Scan
+        if: ${{ (github.event_name == 'push') || (github.event.pull_request.head.repo.owner.login == 'PowerGridModel') }}
```

Comment:
this line is also present in the PGM repo in https://github.com/PowerGridModel/power-grid-model/blob/7e0645bf40cd4c8cacf023b6695f2811d889dd88/.github/workflows/sonar.yml#L76